web/html/src/components/hub/AddTokenMethod.tsx

```diff
@@ -0,0 +1,262 @@
+          <Button
+            id="issue-btn"
+            icon="fa-plus"
+            className="btn-default"
+            handler={() => this.setState({ createRequest: { type: TokenType.ISSUED } })}
+            text={t("Issue a new token")}
+          />
+        );
+
+      case AddTokenMethod.Store:
+        return (
+          <Button
+            id="save-btn"
+            icon="fa-save"
+            className="btn-default"
+            handler={() => this.setState({ createRequest: { type: TokenType.CONSUMED } })}
+            text={t("Store an external token")}
+          />
+        );
+
+      case AddTokenMethod.IssueAndStore:
+        return (
+          <DropdownButton
+            text={t("Add token")}
+            icon="fa-plus"
+            title={t("Add a new access token")}
+            className="btn-primary"
+            items={[
+              <LinkButton
+                className="btn-link"
+                text={t("Issue a new token")}
+                icon="fa-plus"
+                handler={() => this.setState({ createRequest: { type: TokenType.ISSUED } })}
+              />,
+              <LinkButton
+                className="btn-link"
+                text={t("Store an external token")}
+                icon="fa-save"
+                handler={() => this.setState({ createRequest: { type: TokenType.CONSUMED } })}
+              />,
+            ]}
+          />
+        );
+    }
+  }
+
+  private renderCreationForm(): React.ReactNode {
+    if (this.state.createRequest === undefined) {
+      return <></>;
+    }
+
+    return (
+      <Dialog
+        id="creation-modal"
+        title={
+          this.state.createRequest.type === TokenType.ISSUED ? t("Issue a new token") : t("Store an external token")
+        }
+        isOpen={this.state.createRequest !== undefined}
+        onClose={() => this.setState({ createRequest: undefined })}
+        content={
+          <Form model={this.state.createRequest} onValidate={(valid) => this.setState({ createRequestValid: valid })}>
+            <Text
+              name="fqdn"
+              label={t("Server FQDN")}
+              required
+              placeholder={t("e.g. server.domain.com")}
+              labelClass="col-md-3"
+              divClass="col-md-6"
+              validators={[Validation.matches(/^[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)*$/)]}
+              invalidHint={t("Has to be a valid FQDN address")}
+            />
+            {this.state.createRequest.type === TokenType.CONSUMED && (
+              <Text
+                name="token"
+                label={t("Token")}
+                required
+                labelClass="col-md-3"
+                divClass="col-md-6"
+                placeholder="eyJhbGci..."
+              />
+            )}
+          </Form>
+        }
+        footer={
+          <div className="col-lg-6">
+            <div className="pull-right btn-group">
+              <Button
+                id={`creation-modal-cancel-button`}
```

Comment:
Jfyi here and elsewhere, this could also be

```suggestion
                id="creation-modal-cancel-button"
```

Reply:
fixed!